java-extras/src/main/java/org/triplea/yaml/YamlUtils.java

```diff
@@ -0,0 +1,60 @@
+package org.triplea.yaml;
+
+import com.google.common.base.Strings;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Optional;
+import lombok.experimental.UtilityClass;
+import org.snakeyaml.engine.v2.api.Dump;
+import org.snakeyaml.engine.v2.api.DumpSettings;
+import org.snakeyaml.engine.v2.api.Load;
+import org.snakeyaml.engine.v2.api.LoadSettings;
+import org.snakeyaml.engine.v2.exceptions.YamlEngineException;
+
+/**
+ * Methods useful for working reading and writing YAML data. This class largely is a wrapper around
+ * snake yaml.
+ */
+@UtilityClass
+public class YamlUtils {
+  public <T> T readYaml(final String inputString) {
+    final var inputStream =
+        new ByteArrayInputStream(Strings.nullToEmpty(inputString).getBytes(StandardCharsets.UTF_8));
```

Comment:
Does nullToEmpty really make sense here?
If it was null, wouldn't the parser throw an exception due to missing data?
So a precondition would be a better choice IMO

Reply:
This is lifted from: 

```
@UtilityClass
public class StringToInputStream {

  /** Converts a {@code String} to an equivalent {@code InputStream}. */
  public static InputStream asInputStream(final String inputString) {
    return new ByteArrayInputStream(
        Strings.nullToEmpty(inputString).getBytes(StandardCharsets.UTF_8));
  }
}
```
That is in the test-common package. Does it make sense to make that a java-common utility at this point, to support one non-test code usage? I don't know. Keeping that code similar looking in the meantime has value if we do consolidate it in the future.

With that said, a Precondition would not give a user friendly error message and would be a difficult exception type for calling code to deal with. In part the wrapper here is unifying the error scenarios to throw one exception type. An empty or a null input is potentially a valid empty YAML. 

Is handling a null input a silent failure? It's hard to say at this level of code, it could be, but maybe not - it is a smell.